arbiter-core/src/environment/mod.rs

```diff
@@ -668,9 +685,19 @@ impl EventBroadcaster {
 
     /// Loop through each sender and send  `Vec<Log>` emitted from a transaction
     /// downstream to any and all receivers
-    fn broadcast(&self, logs: Vec<Log>) -> Result<(), EnvironmentError> {
-        for sender in &self.0 {
-            sender.send(logs.clone())?;
+    fn broadcast(&self, logs: Option<Vec<Log>>, stop_signal: bool) -> Result<(), EnvironmentError> {
+        if stop_signal {
+            for sender in &self.0 {
+                sender.send(Broadcast::StopSignal)?;
+            }
+            return Ok(());
+        } else {
+            if logs.is_none() {
+                panic!("This shouldn't happen, but we should probably make this an error. Somehow we got told to broadcast `None` logs!");
```

Comment:
lol is this possible to hit?

Reply:
I really don't think so. I think if no logs, you actually broadcast `vec![]`. That could actually be optimized out...